### PR TITLE
Replace deprecated usages of boost

### PR DESCRIPTION
**Pull request by timonegk**
* `boost::progress_display` is deprecated and replaced by `boost::timer::progress_display`
* `std::shared_ptr` can be used instead of `boost::shared_ptr` because the project is using C++ >= 11

### CODE REVIEW

**Comment by timonegk** [comment]
Apparently the change is not in Ubuntu focal so this has to wait until focal is EOL.

File: include/reach/plugins/boost_progress_console_logger.h [hunk 1]
```diff
@@ -18,8 +18,7 @@
 
 #include <reach/interfaces/logger.h>
 
-#include <boost/shared_ptr.hpp>
-#include <boost/progress.hpp>
+#include <boost/timer/progress_display.hpp>
```
Comment (marip8) [inline include/reach/plugins/boost_progress_console_logger.h:21]: I think we could probably do something like this to continue supporting `focal` but fix the deprecation problem

```c++
#if __has_include(<boost/timer/progress_display.hpp>)
#include <boost/timer/progress_display.hpp>
#else
#include <boost/progress.hpp>
#endif
```

Comment (timonegk) [reply include/reach/plugins/boost_progress_console_logger.h:21]: The namespace also changed, so the fix is not as easy, unfortunately

**Comment by marip8** [comment]
Closing in favor of #84; thanks for taking a look at this though

**Pull request closed without merging**